py/desispec/workflow/processing.py

```diff
@@ -1285,7 +1297,7 @@ def recursive_submit_failed(rown, proc_table, submits, id_to_row_map, ptab_name=
         for idep in np.sort(np.atleast_1d(ideps)):
             if idep not in id_to_row_map:
                 if idep // 1000 != row['INTID'] // 1000:
-                    log.info(f"Internal ID: {idep} not in id_to_row_map. "
+                    log.debug(f"Internal ID: {idep} not in id_to_row_map. "
                              + "This is expected since it's from another day. ")
```

Comment:
We've had cases where unprinted debugging log messages inside loops were contributing a significant fraction to the total runtime because the string formatting was happening before being passed into `log.debug`, instead of letting `log.debug` decide whether the message will be printed and only doing the formatting if necessary.  That slowdown might not apply here, but in general it is a better practice to format `log.debug` messages like
```python
log.debug("Internal ID: %d not in id_to_row_map...", idep)
# note: log.debug("message %d", value) not log.debug("message %d" % value)
```
instead of
```
log.debug(f"Internal ID: {idep} not in id_to_row_map...")
```
I prefer the readability of the second form for messages that are going to get printed anyway (info and above), but for debug-level let's use the style that lets the logger decide whether it will format the string or not.

Reply:
Thanks for catching that, this was originally an info-level comment, which is why it used f-strings. When I switched it to a debug comment in this branch I should have switched it. This only runs up to ~2-5 times ~1-30 times per top-level script execution, so I don't think it would have ever made a significant performance hit, but it's worth being consistent with debug-level comments.